pkg/storage-init/storage-init.sh

```diff
@@ -155,7 +157,22 @@ for BLK_DEVICE in $BLK_DEVICES; do
 done
 
 #Recording SMART details to a file
-smartctl -a "$(grep -m 1 /persist < /proc/mounts | cut -d ' ' -f 1)" --json > $SMART_DETAILS_FILE
+SMART_JSON=$(smartctl -a "$(grep -m 1 /persist < /proc/mounts | cut -d ' ' -f 1)" --json)
+CURRENT_POWER_CYCLE_COUNT=$(echo "$SMART_JSON" | grep -o '\bpower_cycle_count[^,]*' | cut -d ':' -f 2 | sed -e 's/^[[:space:]]*//')
+if [ -z "$CURRENT_POWER_CYCLE_COUNT" ]; then
+  CURRENT_POWER_CYCLE_COUNT=0
```

Comment:
I would pick -1 to represent "I don't know"; didn't you use that in the API?

Reply:
Done.